lean4-infoview/src/infoview/info.tsx

```diff
@@ -281,6 +286,7 @@ function InfoAux(props: InfoProps) {
             const [goals, termGoal] = await allReq;
             setGoals(goals);
             setTermGoal(termGoal);
+            onError(undefined)
```

Comment:
What's the purpose of this call? `onError` is meant to be called, well, on error. But it seems like it's used to clear some state here?

Reply:
yeah, it's a hack needed because the implementation of useEventResult is stateful.  I suppose we could handle it in main.tsx when we get a serverRestarted we could call setT on the state for serverStopped, but the setT function is hidden inside useEventResult so it gets a bit messy... it was easier to just do this...   but let us know if there's a cleaner way to do it.

Reply:
Hmm, could you please elaborate on exactly which instance of `useEventResult` is causing trouble? If it is, then it sounds like there might indeed be a deeper issue.

Reply:
Never mind I got my wires crossed, I'm referring to something else, namely, the line in `infoview.ts` where it calls `await this.webviewPanel?.api.serverStopped('');`  The info.tsx change I'm not so sure about, @marianaalanis93 perhaps you should take this one :-)

Reply:
The reason we need to mark this onError as 'undefined' is that if the server dies indefinitely, a dict is generated with the generated error ({error here}). If we restart it later, this variable is left with an empty dict ({}), that's why we have to force it to undefined (as it was originally and before the server had any problem). It should be a problem at the moment we want to handle errors in main.tsx

Reply:
Hm, I see. So I think this was quite confusing to me, because in my mind the `Info` component is supposed to maintain the invariant that the server is running. In particular, with the change in this PR when the server is stopped, in `main.tsx` we return the error message `Server unavailable. Please restart it.` and do not return `<Infos />` at all. This means that all the `<Info>` components will be unmounted and remounted when the server starts again, so they should not persist state, including the `error` state here.

Reply:
Wojciech is right here, we never get to `Infos` with a stopped server.

Reply:
Looks to me like this is clearing a possible error state created on line 297 in the catch block so that an error is not "sticky" and always passed to the <InfoDisplay constructor on line 306.

Reply:
I think I got a bit sidetracked by the `<Infos>` can't be reached if the server is stopped comment.

> If we restart it later, this variable is left with an empty dict ({}), that's why we have to force it to undefined (as it was originally and before the server had any problem).

Did it cause any issues that `error` was set to `{}`?  The idea is that the `error` variable is completely ignored if the status is not `'error'`.  And the end of the `triggerUpdate` function sets the status to `'ready'`.

If we want to reset the `error` variable, then we should just call `setError(undefined)` directly.  There's no need to complicate the `onError` function for this.  (Because the `onError` function is also meant to transition into the `'error'` state.)

Reply:
I agree `setError(undefined)` is better here.  Mariana, can you try that?